src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts

```diff
@@ -728,10 +728,14 @@ export class TerminalTaskSystem extends Disposable implements ITaskSystem {
 		this.collectTaskVariables(variables, task);
 		const resolvedVariables = this.acquireInput(systemInfo, workspaceFolder, task, variables, alreadyResolved);
 
-		return resolvedVariables.then((resolvedVariables) => {
+		return resolvedVariables.then(async (resolvedVariables) => {
 			if (resolvedVariables && !this.isTaskEmpty(task)) {
 				this.currentTask.resolvedVariables = resolvedVariables;
-				return this.executeInTerminal(task, trigger, new VariableResolver(workspaceFolder, systemInfo, resolvedVariables.variables, this.configurationResolverService), workspaceFolder);
+				const resolver = new VariableResolver(workspaceFolder, systemInfo, resolvedVariables.variables, this.configurationResolverService);
```

Comment:
Is the old, possibly wrong, workspace not still getting passed into the VariableResolver?

Reply:
To resolve variables you need a workspace folder. To get the workspace folder you need to resolve the value of `cwd`. I don't see a great solution here.

Reply:
Additionally, the `cwd` doesn't have to be a workspace folder. It could be any folder at all including a folder outside of the workspace. I'm not sure if this would be a problem.

Reply:
Yes you've summarized the problem that I was stumped by.

Reply:
This is probably why I never fixed this :). It's probably not worth re-writing the whole tasks system over when there's a good workaround: use folder tasks.